x/concentrated-liquidity/genesis.go

```diff
@@ -79,6 +79,18 @@ func (k Keeper) InitGenesis(ctx sdk.Context, genState genesis.GenesisState) {
 		feePositionKey := types.KeyFeePositionAccumulator(positionWrapper.Position.PositionId)
 
 		k.initOrUpdateAccumPosition(ctx, feeAccumObject, positionWrapper.FeeAccumRecord.InitAccumValue, feePositionKey, positionWrapper.FeeAccumRecord.NumShares, positionWrapper.FeeAccumRecord.UnclaimedRewards, positionWrapper.FeeAccumRecord.Options)
+		uptimeAccumObject, err := k.GetUptimeAccumulators(ctx, positionWrapper.Position.PoolId)
+		if err != nil {
+			panic(err)
+		}
+		uptimePositionKey := string(types.KeyPositionId(positionWrapper.Position.PositionId))
+
+		// for each position we initialize all uptime  accumulator
+		for _, uptimeRecord := range positionWrapper.UptimeAccumRecord {
+			for _, uptimeAccum := range uptimeAccumObject {
+				k.initOrUpdateAccumPosition(ctx, uptimeAccum, uptimeRecord.InitAccumValue, uptimePositionKey, uptimeRecord.NumShares, uptimeRecord.UnclaimedRewards, uptimeRecord.Options)
+			}
+		}
```

Comment:
> Each exported position record should be associated with one uptime only.

the way i thought of this was, each position can have multiple uptime records associated with it. So the outer loop checks for the position and the inner loop checks for all the uptime that could be associated with the position

Reply:
Assume we have 1 position exported with 3 uptimes for simplicity (1ns, 1s and 1min)

Then, the `positionWrapper.UptimeAccumRecord` list here consists of 3 records, A (1ns), B(1s) and C (1min)

What ends up happening is that for all of A, B and C we initialize records with 1ns, 1s and 1min.

I believe that we end up having just 3 records C(1ns), C(1s) and C (1min) because the last outer loop iteration overwrites

Lmk if that makes sense

Reply:
I thought we initialize 
A(1ns) A(1s) A(1min) B(1ns) B(1s) B(1min)  C(1ns) C(1s) C(1min) 

ooh so the inner loop should be outside? so that we initialize an accumPosition for every `uptimeRecord` for every `uptimeAccum`, am i correctly understanding this?

Reply:
> I thought we initialize
A(1ns) A(1s) A(1min) B(1ns) B(1s) B(1min) C(1ns) C(1s) C(1min)

This would be the case if we had 3 positions.

In my example, I had 1 position with 3 records exported corresponding to every uptime A (1ns), B (1s) and C(1min)

> ooh so the inner loop should be outside?

Let's take a step back from the inner vs outer loop.

When exporting genesis, we can export a matrix `recordsForEachUptime` [][]Record where every element is a slice of all records for a supported uptime at index `i`.

That way, during import, when iterating over `recordsForEachUptime` we can attach all records to a specific uptime accumulator that we know those positions belong to.

In the current approach, we lose the context of which uptime accumulator a position belongs to by exporting it into a slice from the start